pybamm/discretisations/base_discretisation.py

```diff
@@ -533,11 +508,13 @@ def check_model(self, model):
         )
 
         # Check variables in variable list against rhs
+        # Be lenient with size check if the variable in model.variables is broadcasted
         for var in model.rhs.keys():
             if var.name in model.variables.keys():
-                assert (
-                    model.rhs[var].evaluate(0, y0).shape
-                    == model.variables[var.name].evaluate(0, y0).shape
+                assert model.rhs[var].evaluate(0, y0).shape == model.variables[
+                    var.name
+                ].evaluate(0, y0).shape or isinstance(
```

Comment:
why is this necessary? shouldn't the discretisation replace all the Broadcast nodes with NumpyBroadcast?

Reply:
changing to NumpyBroadcast